src/Hpack/Render/Dependency.hs

```diff
@@ -0,0 +1,49 @@
+module Hpack.Render.Dependency (
+  renderDependencies
+, renderVersion
+) where
+
+import           Data.DList (DList)
+import qualified Data.DList as DList
+import           Data.Foldable (toList)
+import qualified Data.Map as Map
+import           Data.Semigroup (Semigroup(..))
+import           Hpack.Render.Dsl
+import           Hpack.Syntax.Dependency
+import           Hpack.Syntax.DependencyVersion
+
+data DependencyFields = DependencyFields {
+  _dependencyFieldDepends :: DList String
+, _dependencyFieldMixins :: DList String
```

Comment:
Does the `DList` actually makes a difference here?  Does `foldMapWithKey` give us left associative concatenation?